---
title: >
 The `iSEEfier` User's Guide
author:
- name: Najla Abassi
  affiliation: 
  - Institute of Medical Biostatistics, Epidemiology and Informatics (IMBEI), Mainz
  email: najla.abassi@uni-mainz.de
- name: Federico Marini
  affiliation: 
  - Institute of Medical Biostatistics, Epidemiology and Informatics (IMBEI), Mainz
  - Research Center for Immunotherapy (FZI), Mainz
  email: marinif@uni-mainz.de
date: "`r BiocStyle::doc_date()`"
output: 
  BiocStyle::html_document:
    toc: true
    toc_float: true
vignette: >
  %\VignetteIndexEntry{iSEEfier_userguide}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
  %\VignettePackage{iSEEfier}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  error    = FALSE,
  warning  = FALSE,
  eval     = TRUE,
  message  = FALSE
)
```

# Introduction {#introduction}

This vignette describes how to use `r BiocStyle::Biocpkg("iSEEfier")` package to configure various initial states of iSEE instances, in order to simplify the task of visualizing RNAseq data in `r BiocStyle::Biocpkg("iSEE")`.

In the remainder of this vignette, we will illustrate the main features of `r BiocStyle::Biocpkg("iSEEfier")` on a publicly available dataset from Baron et al. "A Single-Cell Transcriptomic Map of the Human and Mouse Pancreas Reveals Inter- and Intra-cell Population Structure", published in Cell Systems in 2016. [doi:10.1016/j.cels.2016.08.011](https://doi.org/10.1016/j.cels.2016.08.011).

The data is made available via the `r BiocStyle::Biocpkg("scRNAseq")` Bioconductor package.
We'll simply use the mouse dataset, consisting of islets isolated from five C57BL/6 and ICR mice.

# Getting started {#gettingstarted}

To install `r BiocStyle::Biocpkg("iSEEfier")` package, we start R and enter:

```{r install, eval=FALSE}
if (!requireNamespace("BiocManager", quietly = TRUE))
  install.packages("BiocManager")

BiocManager::install("iSEEfier")
```

Once installed, the package can be loaded and attached to the current workspace as follows:

```{r setup}
library("iSEEfier")
```

# Create an initial state for gene expression visualization using `iSEEinit()`

When we have all input elements ready, we can create an iSEE initial state by running:

```{r runfunc, eval=FALSE}
iSEEinit(sce = sce_obj,
         feature.list = feature_list,
         reddim.type = reduced_dim,
         clusters = cluster,
         groups =  group,
         markdownboard = FALSE,
         dynamicMarkerTable = FALSE)
```

To configure the initial state of our iSEE instance using `iSEEfier`, we need five parameters:

1.  `sce` : A `SingleCellExperiment` object. This object stores information of different quantifications (counts, log-expression...), dimensionality reduction coordinates (t-SNE, UMAP...), as well as some metadata related to the samples and features.

We'll start by loading the `sce` object

```{r loaddata}
library("scRNAseq")
sce <- BaronPancreasData('mouse')
sce
```

Let's add the normalized counts

```{r logNorm}
library("scuttle")
sce <- logNormCounts(sce)
```

Now we can add different dimensionality reduction coordinates

```{r reddim}
library("scater")
sce <- runPCA(sce)
sce <- runTSNE(sce)
sce <- runUMAP(sce)
```

We can also directly load the the preprocessed object

```{r save-sce, eval=FALSE}
#> sce <- readRDS("./data/ZhaoImmuneLiverData_processed.RDS") #maybe dont with RDS?
```

Now our `sce` is ready, we can move on to the next argument.

2.  `feature.list` : which is a list of genes/features of interest. Let's say we would like to visualize the expression of some genes that were identified as marker genes for different cell population.

```{r genelist}
gene_list <- c("Gcg", # alpha
               "Ins1") # beta
```

3.  `reddim.type` : In this example we decided to plot our data as a t-SNE plot.

```{r reddim-type}
reddim_type <- "TSNE"
```

4.  `clusters` : Now we specify what clusters/cell-types/states/samples we would like to color/split our data with

```{r cluster-id}
# cell populations
cluster <- "label" #the name should match what's in the colData names
```

5.  `groups` : Here we can add the groups/conditions/cell-types

```{r group-id}
# ICR vs C57BL/6
group <- "strain" #the name should match what's in the colData names
```

At this point, all the elements are ready to be transferred into `iSEEinit()`

```{r initial1}
initial1 <- iSEEinit(sce = sce,
                    feature.list = gene_list,
                    clusters = cluster,
                    groups = group,
                    markdownboard = TRUE,
                    dynamicMarkerTable = TRUE)
```

Now we are one step away from visualizing our list of genes of interest. All that's left to do is to run `iSEE` with the initial state created with `iSEEinit`

```{r iSEEviz1, eval=FALSE}
library("iSEE")
iSEE(sce, initial= initial1)
```

This instance, generated with `iSEEinit()`, returns a combination of panels, linked to each other, with the goal of visualizing the expression of certain marker genes in each cell population/group:

-  A `ReducedDimensionPlot`, `FeatureAssayPlot` and `RowDataTable` for each single gene.

- A `ComplexHeatmapPlot` with all genes in `feature.list`

- A `DynamicMarkerTable` that identifies marker genes from a sample selection.

- A `ColumnDataPlot`

- A `MarkdownBoard`

# Create an initial state for feature sets exploration using `iSEEnrich()`

Sometimes it is interesting to look at some specific feature sets and the associated genes. That's when the utility of iSEEnrich becomes apparent.
We will need 4 elements to explore feature sets of interest:

- `sce`	: A SingleCellExperiment object

- `collection`	: A character vector specifying the gene set collections of interest (it is possible to use GO or KEGG terms)

- `gene_identifer`	: A character string specifying the identifier to use to extract gene IDs for the organism package. This can be **"ENS"** for ENSEMBL ids, **"SYMBOL"** for gene names...

- `organsim`	: A character string of the `org.*.eg.db` package to use to extract mappings of gene sets to gene IDs.

```{r set-param}
GO_collection <- "GO"
Hs_organism <- "org.Mm.eg.db"
gene_id <- "SYMBOL"
```

Now let's create this initial setup for `iSEE` using `iSEEnrich()`

```{r initial2}
results <- iSEEnrich(sce = sce,
                     collection = GO_collection,
                     organism = Hs_organism,
                     gene_identifer = gene_id)
```

To start the iSEE instance we run:

```{r iSEEviz2,  eval=FALSE}
iSEE(results$sce1, initial = results$initial)
```

# Visualize a preview of the initial configurations with `iSEEconfiviewer()`

Previously, we successfully generated two distinct initial configurations for iSEE. However, understanding the expected content of our iSEE instances is not always straightforward. That's when we can use `iSEEconfigviewer()`.

We only need as an input the initial configuration to obtain a graphical visualization of the expected the corresponding iSEE instance:

```{r panelgraph}
library(ggplot2)
iSEEconfigviewer(initial = initial1)
iSEEconfigviewer(initial = results$initial)
```

# Visualize Network connections between panels with `iSEEnetworkviewer()`

As some of these panels are linked to each other, we can visualize these networks with `iSEEnetworkviewer()`. Similar to `iSEEconfigviewer()`, this function takes the initial setup as input:

```{r networkviz}
library("igraph")
g1 <- iSEEnetworkviewer(initial1)
visNetwork::visIgraph(g1) # visualisation for initial1

initial2 <- results$initial
g2 <- iSEEnetworkviewer(initial2)
visNetwork::visIgraph(g2) # visualisation for initial2

```

# Merge different initial configurations with `glue_configs()`

Sometimes, it would be interesting to merge different iSEE initial configurations to visualize all different panel in the same iSEE instance.

```{r glueconfig}
merged_config <- glue_configs(initial1,initial2)
```

We can then preview the content of this initial configuration

```{r preview}
iSEEconfigviewer(merged_config)
```


# Session info {.unnumbered}

```{r}
sessionInfo()
```