<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="iSEEfier">
<title>The `iSEEfier` User's Guide
 • iSEEfier</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="The `iSEEfier` User's Guide
">
<meta property="og:description" content="iSEEfier">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">iSEEfier</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/iSEEfier_userguide.html">The `iSEEfier` User's Guide
</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/NajlaAbassi/iSEEfier/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>The `iSEEfier` User's Guide
</h1>
                        <h4 data-toc-skip class="author">Najla
Abassi</h4>
            <address class="author_afil">
      Institute of Medical Biostatistics, Epidemiology and Informatics
(IMBEI), Mainz<br><a class="author_email" href="mailto:#"></a><a href="mailto:najla.abassi@uni-mainz.de" class="email">najla.abassi@uni-mainz.de</a>
      </address>
                              <h4 data-toc-skip class="author">Federico
Marini</h4>
            <address class="author_afil">
      Institute of Medical Biostatistics, Epidemiology and Informatics
(IMBEI), MainzResearch Center for Immunotherapy (FZI),
Mainz<br><a class="author_email" href="mailto:#"></a><a href="mailto:marinif@uni-mainz.de" class="email">marinif@uni-mainz.de</a>
      </address>
                  
            <h4 data-toc-skip class="date">22 March 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/NajlaAbassi/iSEEfier/blob/HEAD/vignettes/iSEEfier_userguide.Rmd" class="external-link"><code>vignettes/iSEEfier_userguide.Rmd</code></a></small>
      <div class="d-none name"><code>iSEEfier_userguide.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette describes how to use the <em><a href="https://bioconductor.org/packages/3.17/iSEEfier" class="external-link">iSEEfier</a></em>
package to configure various initial states of iSEE instances, in order
to simplify the task of visualizing single-cell RNA-seq, bulk RNA-seq
data, or even your proteomics data in <em><a href="https://bioconductor.org/packages/3.17/iSEE" class="external-link">iSEE</a></em>.</p>
<p>In the remainder of this vignette, we will illustrate the main
features of <em><a href="https://bioconductor.org/packages/3.17/iSEEfier" class="external-link">iSEEfier</a></em>
on a publicly available dataset from Baron et al. “A Single-Cell
Transcriptomic Map of the Human and Mouse Pancreas Reveals Inter- and
Intra-cell Population Structure”, published in Cell Systems in 2016. <a href="https://doi.org/10.1016/j.cels.2016.08.011" class="external-link">doi:10.1016/j.cels.2016.08.011</a>.</p>
<p>The data is made available via the <em><a href="https://bioconductor.org/packages/3.17/scRNAseq" class="external-link">scRNAseq</a></em>
Bioconductor package. We’ll simply use the mouse dataset, consisting of
islets isolated from five C57BL/6 and ICR mice.</p>
</div>
<div class="section level2">
<h2 id="gettingstarted">Getting started<a class="anchor" aria-label="anchor" href="#gettingstarted"></a>
</h2>
<p>To install <em><a href="https://bioconductor.org/packages/3.17/iSEEfier" class="external-link">iSEEfier</a></em>
package, we start R and enter:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"iSEEfier"</span><span class="op">)</span></span></code></pre></div>
<p>Once installed, the package can be loaded and attached to the current
workspace as follows:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/NajlaAbassi/iSEEfier" class="external-link">"iSEEfier"</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="create-an-initial-state-for-gene-expression-visualization-using-iseeinit">Create an initial state for gene expression visualization using
<code>iSEEinit()</code><a class="anchor" aria-label="anchor" href="#create-an-initial-state-for-gene-expression-visualization-using-iseeinit"></a>
</h2>
<p>When we have all input elements ready, we can create an
<code>iSEE</code> initial state by running:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/iSEEinit.html">iSEEinit</a></span><span class="op">(</span>sce <span class="op">=</span> <span class="va">sce_obj</span>,</span>
<span>         features <span class="op">=</span> <span class="va">feature_list</span>,</span>
<span>         reddim.type <span class="op">=</span> <span class="va">reduced_dim</span>,</span>
<span>         clusters <span class="op">=</span> <span class="va">cluster</span>,</span>
<span>         groups <span class="op">=</span> <span class="va">group</span>,</span>
<span>         add_markdown_panel <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>         add_dynamicTable_panel <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>To configure the initial state of our <code>iSEE</code> instance
using <code><a href="../reference/iSEEinit.html">iSEEinit()</a></code>, we need five parameters:</p>
<ol style="list-style-type: decimal">
<li>
<code>sce</code> : A <code>SingleCellExperiment</code> object. This
object stores information of different quantifications (counts,
log-expression…), dimensionality reduction coordinates (t-SNE, UMAP…),
as well as some metadata related to the samples and features.</li>
</ol>
<p>We’ll start by loading the <code>sce</code> object:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"scRNAseq"</span><span class="op">)</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scRNAseq/man/BaronPancreasData.html" class="external-link">BaronPancreasData</a></span><span class="op">(</span><span class="st">'mouse'</span><span class="op">)</span></span>
<span><span class="va">sce</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 14878 1886 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(14878): X0610007P14Rik X0610009B22Rik ... Zzz3 l7Rn6</span></span>
<span><span class="co">#&gt; rowData names(0):</span></span>
<span><span class="co">#&gt; colnames(1886): mouse1_lib1.final_cell_0001 mouse1_lib1.final_cell_0002</span></span>
<span><span class="co">#&gt;   ... mouse2_lib3.final_cell_0394 mouse2_lib3.final_cell_0395</span></span>
<span><span class="co">#&gt; colData names(2): strain label</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span></code></pre></div>
<p>Let’s add the normalized counts</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"scuttle"</span><span class="op">)</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/logNormCounts.html" class="external-link">logNormCounts</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span></code></pre></div>
<p>Now we can add different dimensionality reduction coordinates</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="http://bioconductor.org/packages/scater/" class="external-link">"scater"</a></span><span class="op">)</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocSingular/man/runPCA.html" class="external-link">runPCA</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runTSNE.html" class="external-link">runTSNE</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runUMAP.html" class="external-link">runUMAP</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span></code></pre></div>
<p>Now our <code>sce</code> is ready, we can move on to the next
argument.</p>
<ol start="2" style="list-style-type: decimal">
<li>
<code>features</code> : which is a list or a vector of
genes/features of interest. Let’s say we would like to visualize the
expression of some genes that were identified as marker genes for
different cell population.</li>
</ol>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Gcg"</span>, <span class="co"># alpha</span></span>
<span>               <span class="st">"Ins1"</span><span class="op">)</span> <span class="co"># beta</span></span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>
<code>reddim_type</code> : In this example we decided to plot our
data as a t-SNE plot.</li>
</ol>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">reddim_type</span> <span class="op">&lt;-</span> <span class="st">"TSNE"</span></span></code></pre></div>
<ol start="4" style="list-style-type: decimal">
<li>
<code>clusters</code> : Now we specify what
clusters/cell-types/states/samples we would like to color/split our data
with</li>
</ol>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># cell populations</span></span>
<span><span class="va">cluster</span> <span class="op">&lt;-</span> <span class="st">"label"</span> <span class="co">#the name should match what's in the colData names</span></span></code></pre></div>
<ol start="5" style="list-style-type: decimal">
<li>
<code>groups</code> : Here we can add the
groups/conditions/cell-types</li>
</ol>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># ICR vs C57BL/6</span></span>
<span><span class="va">group</span> <span class="op">&lt;-</span> <span class="st">"strain"</span> <span class="co">#the name should match what's in the colData names</span></span></code></pre></div>
<p>We can choose to include in this initial step a
<code>MarkdownBoard</code> and a <code>DynamicMarkerTable</code>, along
with its linked panels by setting the arguments
<code>add_markdown_panel</code> and <code>add_dynamicTable_panel</code>
to <code>TRUE</code>.</p>
<p>At this point, all the elements are ready to be transferred into
<code><a href="../reference/iSEEinit.html">iSEEinit()</a></code></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">initial1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/iSEEinit.html">iSEEinit</a></span><span class="op">(</span>sce <span class="op">=</span> <span class="va">sce</span>,</span>
<span>                    features <span class="op">=</span> <span class="va">gene_list</span>,</span>
<span>                    clusters <span class="op">=</span> <span class="va">cluster</span>,</span>
<span>                    groups <span class="op">=</span> <span class="va">group</span>,</span>
<span>                    add_markdown_panel <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                    add_dynamicTable_panel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Now we are one step away from visualizing our list of genes of
interest. All that’s left to do is to run <code>iSEE</code> with the
initial state created with <code><a href="../reference/iSEEinit.html">iSEEinit()</a></code></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/iSEE/iSEE" class="external-link">"iSEE"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="va">sce</span>, initial<span class="op">=</span> <span class="va">initial1</span><span class="op">)</span></span></code></pre></div>
<p>This instance, generated with <code><a href="../reference/iSEEinit.html">iSEEinit()</a></code>, returns a
combination of panels, linked to each other, with the goal of
visualizing the expression of certain marker genes in each cell
population/group:</p>
<ul>
<li><p>A <code>ReducedDimensionPlot</code>,
<code>FeatureAssayPlot</code> and <code>RowDataTable</code> for each
single gene in <code>features</code>.</p></li>
<li><p>A <code>ComplexHeatmapPlot</code> with all genes in
<code>features</code></p></li>
<li><p>A <code>DynamicMarkerTable</code> that identifies marker genes
from a sample selection.</p></li>
<li><p>A <code>ColumnDataPlot</code> panel</p></li>
<li><p>A <code>MarkdownBoard</code> panel</p></li>
</ul>
</div>
<div class="section level2">
<h2 id="create-an-initial-state-for-feature-sets-exploration-using-iseenrich">Create an initial state for feature sets exploration using
<code>iSEEnrich()</code><a class="anchor" aria-label="anchor" href="#create-an-initial-state-for-feature-sets-exploration-using-iseenrich"></a>
</h2>
<p>Sometimes it is interesting to look at some specific feature sets and
the associated genes. That’s when the utility of <code>iSEEnrich</code>
becomes apparent. We will need 4 elements to explore feature sets of
interest:</p>
<ul>
<li><p><code>sce</code>: A SingleCellExperiment object</p></li>
<li><p><code>collection</code>: A character vector specifying the gene
set collections of interest (it is possible to use GO or KEGG
terms)</p></li>
<li><p><code>gene_identifier</code>: A character string specifying the
identifier to use to extract gene IDs for the organism package. This can
be <strong>“ENS”</strong> for ENSEMBL ids, <strong>“SYMBOL”</strong> for
gene names…</p></li>
<li><p><code>organism</code>: A character string of the
<code>org.*.eg.db</code> package to use to extract mappings of gene sets
to gene IDs.</p></li>
</ul>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GO_collection</span> <span class="op">&lt;-</span> <span class="st">"GO"</span></span>
<span><span class="va">Mm_organism</span> <span class="op">&lt;-</span> <span class="st">"org.Mm.eg.db"</span></span>
<span><span class="va">gene_id</span> <span class="op">&lt;-</span> <span class="st">"SYMBOL"</span></span></code></pre></div>
<p>Now let’s create this initial setup for <code>iSEE</code> using
<code><a href="../reference/iSEEnrich.html">iSEEnrich()</a></code></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/iSEEnrich.html">iSEEnrich</a></span><span class="op">(</span>sce <span class="op">=</span> <span class="va">sce</span>,</span>
<span>                     collection <span class="op">=</span> <span class="va">GO_collection</span>,</span>
<span>                     organism <span class="op">=</span> <span class="va">Mm_organism</span>,</span>
<span>                     gene_identifier <span class="op">=</span> <span class="va">gene_id</span><span class="op">)</span></span></code></pre></div>
<p><code>iSEEnrich</code> will specifically return a list with the
updated <code>sce</code> object and its associated <code>initial</code>
configuration. To start the <code>iSEE</code> instance we run:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">sce</span>, initial <span class="op">=</span> <span class="va">results</span><span class="op">$</span><span class="va">initial</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="visualize-a-preview-of-the-initial-configurations-with-view_initial_tiles">Visualize a preview of the initial configurations with
<code>view_initial_tiles()</code><a class="anchor" aria-label="anchor" href="#visualize-a-preview-of-the-initial-configurations-with-view_initial_tiles"></a>
</h2>
<p>Previously, we successfully generated two distinct initial
configurations for iSEE. However, understanding the expected content of
our iSEE instances is not always straightforward. That’s when we can use
<code><a href="../reference/view_initial_tiles.html">view_initial_tiles()</a></code>.</p>
<p>We only need as an input the initial configuration to obtain a
graphical visualization of the expected the corresponding
<code>iSEE</code> instance:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/view_initial_tiles.html">view_initial_tiles</a></span><span class="op">(</span>initial <span class="op">=</span> <span class="va">initial1</span><span class="op">)</span></span></code></pre></div>
<p><img src="iSEEfier_userguide_files/figure-html/panelgraph-1.png" width="700"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/view_initial_tiles.html">view_initial_tiles</a></span><span class="op">(</span>initial <span class="op">=</span> <span class="va">results</span><span class="op">$</span><span class="va">initial</span><span class="op">)</span></span></code></pre></div>
<p><img src="iSEEfier_userguide_files/figure-html/panelgraph-2.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="visualize-network-connections-between-panels-with-view_initial_network">Visualize network connections between panels with
<code>view_initial_network()</code><a class="anchor" aria-label="anchor" href="#visualize-network-connections-between-panels-with-view_initial_network"></a>
</h2>
<p>As some of these panels are linked to each other, we can visualize
these networks with <code><a href="../reference/view_initial_network.html">view_initial_network()</a></code>. Similar to
<code>iSEEconfigviewer()</code>, this function takes the initial setup
as input:</p>
<p>This function always returns the <code>igraph</code> object
underlying the visualizations that can be displayed as a side
effect.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://r.igraph.org/" class="external-link">"igraph"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://datastorm-open.github.io/visNetwork/" class="external-link">"visNetwork"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">g1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/view_initial_network.html">view_initial_network</a></span><span class="op">(</span><span class="va">initial1</span>, plot_format <span class="op">=</span> <span class="st">"igraph"</span><span class="op">)</span></span></code></pre></div>
<p><img src="iSEEfier_userguide_files/figure-html/networkviz-1.png" width="700"></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">g1</span></span>
<span><span class="co">#&gt; IGRAPH c4c3491 DN-- 14 5 -- </span></span>
<span><span class="co">#&gt; + attr: name (v/c), color (v/c)</span></span>
<span><span class="co">#&gt; + edges from c4c3491 (vertex names):</span></span>
<span><span class="co">#&gt; [1] ReducedDimensionPlot1-&gt;ColumnDataPlot1      </span></span>
<span><span class="co">#&gt; [2] ReducedDimensionPlot2-&gt;ColumnDataPlot1      </span></span>
<span><span class="co">#&gt; [3] ReducedDimensionPlot3-&gt;FeatureAssayPlot3    </span></span>
<span><span class="co">#&gt; [4] ReducedDimensionPlot4-&gt;FeatureAssayPlot4    </span></span>
<span><span class="co">#&gt; [5] DynamicMarkerTable1  -&gt;ReducedDimensionPlot4</span></span>
<span></span>
<span><span class="va">initial2</span> <span class="op">&lt;-</span> <span class="va">results</span><span class="op">$</span><span class="va">initial</span></span>
<span><span class="va">g2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/view_initial_network.html">view_initial_network</a></span><span class="op">(</span><span class="va">initial2</span>, plot_format <span class="op">=</span> <span class="st">"visNetwork"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="merge-different-initial-configurations-with-glue_initials">Merge different initial configurations with
<code>glue_initials()</code><a class="anchor" aria-label="anchor" href="#merge-different-initial-configurations-with-glue_initials"></a>
</h2>
<p>Sometimes, it would be interesting to merge different
<code>iSEE</code> initial configurations to visualize all different
panel in the same <code>iSEE</code> instance.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">merged_config</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/glue_initials.html">glue_initials</a></span><span class="op">(</span><span class="va">initial1</span>,<span class="va">initial2</span><span class="op">)</span></span></code></pre></div>
<p>We can then preview the content of this initial configuration</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/view_initial_tiles.html">view_initial_tiles</a></span><span class="op">(</span><span class="va">merged_config</span><span class="op">)</span></span></code></pre></div>
<p><img src="iSEEfier_userguide_files/figure-html/preview-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="related-work">Related work<a class="anchor" aria-label="anchor" href="#related-work"></a>
</h2>
<p>The idea of launching <code><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html" class="external-link">iSEE()</a></code> with some specific
configuration is not entirely new, and it was covered in some use cases
by the <code>mode_</code> functions available in the <em><a href="https://bioconductor.org/packages/3.17/iSEEu" class="external-link">iSEEu</a></em>
package.<br>
There, the user has access to the following:</p>
<ul>
<li>
<code><a href="https://rdrr.io/pkg/iSEEu/man/modeEmpty.html" class="external-link">iSEEu::modeEmpty()</a></code> - this will launch <code>iSEE</code>
without any panels, and let you build up the configuration from the
scratch. Easy to start, easy to build.</li>
<li>
<code><a href="https://rdrr.io/pkg/iSEEu/man/modeGating.html" class="external-link">iSEEu::modeGating()</a></code> - this will open <code>iSEE</code>
with multiple chain-linked FeatureExpressionPlot panels, just like when
doing some in silico gating. This could be a very good fit if working
with mass cytometry data.</li>
<li>
<code><a href="https://rdrr.io/pkg/iSEEu/man/modeReducedDim.html" class="external-link">iSEEu::modeReducedDim()</a></code> - <code>iSEE</code> will be
ready to compare multiple ReducedDimensionPlot panels, which is a
suitable option to compare the views resulting from different embeddings
(and/or embeddings generated with slightly different parameter
configurations).</li>
</ul>
<p>The <code>mode</code>s directly launch an instance of
<code>iSEE</code>, whereas the functionality in <em><a href="https://bioconductor.org/packages/3.17/iSEEfier" class="external-link">iSEEfier</a></em>
is rather oriented to obtain more tailored-to-the-data-at-hand
<code>initial</code> objects, that can subsequently be passed as an
argument to the <code><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html" class="external-link">iSEE()</a></code> call.</p>
<p>We encourage users to submit suggestions about their “classical ways”
of using <code>iSEE</code> on their data - be that by opening an issue
or already proposing a Pull Request on GitHub.</p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">#&gt; Platform: x86_64-apple-darwin20 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: macOS Ventura 13.6</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Europe/Berlin</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] visNetwork_2.1.2            igraph_1.5.1               </span></span>
<span><span class="co">#&gt;  [3] scater_1.28.0               ggplot2_3.4.4              </span></span>
<span><span class="co">#&gt;  [5] scuttle_1.10.3              scRNAseq_2.14.0            </span></span>
<span><span class="co">#&gt;  [7] SingleCellExperiment_1.22.0 SummarizedExperiment_1.30.2</span></span>
<span><span class="co">#&gt;  [9] Biobase_2.60.0              GenomicRanges_1.52.1       </span></span>
<span><span class="co">#&gt; [11] GenomeInfoDb_1.36.4         IRanges_2.34.1             </span></span>
<span><span class="co">#&gt; [13] S4Vectors_0.38.2            BiocGenerics_0.46.0        </span></span>
<span><span class="co">#&gt; [15] MatrixGenerics_1.12.3       matrixStats_1.0.0          </span></span>
<span><span class="co">#&gt; [17] iSEEfier_0.99.0             BiocStyle_2.28.1           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] splines_4.3.1                 later_1.3.1                  </span></span>
<span><span class="co">#&gt;   [3] BiocIO_1.10.0                 bitops_1.0-7                 </span></span>
<span><span class="co">#&gt;   [5] filelock_1.0.2                tibble_3.2.1                 </span></span>
<span><span class="co">#&gt;   [7] XML_3.99-0.15                 lifecycle_1.0.3              </span></span>
<span><span class="co">#&gt;   [9] doParallel_1.0.17             rprojroot_2.0.3              </span></span>
<span><span class="co">#&gt;  [11] lattice_0.22-5                ensembldb_2.24.1             </span></span>
<span><span class="co">#&gt;  [13] magrittr_2.0.3                sass_0.4.7                   </span></span>
<span><span class="co">#&gt;  [15] rmarkdown_2.25                jquerylib_0.1.4              </span></span>
<span><span class="co">#&gt;  [17] yaml_2.3.7                    httpuv_1.6.12                </span></span>
<span><span class="co">#&gt;  [19] DBI_1.1.3                     RColorBrewer_1.1-3           </span></span>
<span><span class="co">#&gt;  [21] abind_1.4-5                   zlibbioc_1.46.0              </span></span>
<span><span class="co">#&gt;  [23] Rtsne_0.16                    purrr_1.0.2                  </span></span>
<span><span class="co">#&gt;  [25] AnnotationFilter_1.24.0       RCurl_1.98-1.13              </span></span>
<span><span class="co">#&gt;  [27] rappdirs_0.3.3                circlize_0.4.15              </span></span>
<span><span class="co">#&gt;  [29] GenomeInfoDbData_1.2.10       ggrepel_0.9.4                </span></span>
<span><span class="co">#&gt;  [31] irlba_2.3.5.1                 pkgdown_2.0.7                </span></span>
<span><span class="co">#&gt;  [33] iSEEhex_1.2.0                 DelayedMatrixStats_1.22.6    </span></span>
<span><span class="co">#&gt;  [35] codetools_0.2-19              DelayedArray_0.26.7          </span></span>
<span><span class="co">#&gt;  [37] DT_0.30                       xml2_1.3.5                   </span></span>
<span><span class="co">#&gt;  [39] tidyselect_1.2.0              shape_1.4.6                  </span></span>
<span><span class="co">#&gt;  [41] farver_2.1.1                  viridis_0.6.4                </span></span>
<span><span class="co">#&gt;  [43] ScaledMatrix_1.8.1            shinyWidgets_0.8.0           </span></span>
<span><span class="co">#&gt;  [45] BiocFileCache_2.8.0           GenomicAlignments_1.36.0     </span></span>
<span><span class="co">#&gt;  [47] jsonlite_1.8.7                GetoptLong_1.0.5             </span></span>
<span><span class="co">#&gt;  [49] BiocNeighbors_1.18.0          ellipsis_0.3.2               </span></span>
<span><span class="co">#&gt;  [51] iterators_1.0.14              systemfonts_1.0.5            </span></span>
<span><span class="co">#&gt;  [53] foreach_1.5.2                 tools_4.3.1                  </span></span>
<span><span class="co">#&gt;  [55] progress_1.2.2                ragg_1.2.6                   </span></span>
<span><span class="co">#&gt;  [57] Rcpp_1.0.11                   glue_1.6.2                   </span></span>
<span><span class="co">#&gt;  [59] gridExtra_2.3                 xfun_0.42                    </span></span>
<span><span class="co">#&gt;  [61] mgcv_1.9-0                    dplyr_1.1.3                  </span></span>
<span><span class="co">#&gt;  [63] shinydashboard_0.7.2          withr_2.5.2                  </span></span>
<span><span class="co">#&gt;  [65] BiocManager_1.30.22           fastmap_1.1.1                </span></span>
<span><span class="co">#&gt;  [67] fansi_1.0.5                   shinyjs_2.1.0                </span></span>
<span><span class="co">#&gt;  [69] rsvd_1.0.5                    digest_0.6.33                </span></span>
<span><span class="co">#&gt;  [71] R6_2.5.1                      mime_0.12                    </span></span>
<span><span class="co">#&gt;  [73] textshaping_0.3.7             colorspace_2.1-0             </span></span>
<span><span class="co">#&gt;  [75] biomaRt_2.56.1                RSQLite_2.3.2                </span></span>
<span><span class="co">#&gt;  [77] utf8_1.2.4                    generics_0.1.3               </span></span>
<span><span class="co">#&gt;  [79] hexbin_1.28.3                 FNN_1.1.3.2                  </span></span>
<span><span class="co">#&gt;  [81] rtracklayer_1.60.1            prettyunits_1.2.0            </span></span>
<span><span class="co">#&gt;  [83] httr_1.4.7                    htmlwidgets_1.6.2            </span></span>
<span><span class="co">#&gt;  [85] S4Arrays_1.2.0                org.Mm.eg.db_3.17.0          </span></span>
<span><span class="co">#&gt;  [87] uwot_0.1.16                   iSEE_2.12.0                  </span></span>
<span><span class="co">#&gt;  [89] pkgconfig_2.0.3               gtable_0.3.4                 </span></span>
<span><span class="co">#&gt;  [91] blob_1.2.4                    ComplexHeatmap_2.16.0        </span></span>
<span><span class="co">#&gt;  [93] XVector_0.40.0                htmltools_0.5.6.1            </span></span>
<span><span class="co">#&gt;  [95] bookdown_0.36                 ProtGenerics_1.32.0          </span></span>
<span><span class="co">#&gt;  [97] rintrojs_0.3.3                clue_0.3-65                  </span></span>
<span><span class="co">#&gt;  [99] scales_1.2.1                  png_0.1-8                    </span></span>
<span><span class="co">#&gt; [101] knitr_1.45                    rstudioapi_0.15.0            </span></span>
<span><span class="co">#&gt; [103] rjson_0.2.21                  nlme_3.1-163                 </span></span>
<span><span class="co">#&gt; [105] curl_5.1.0                    shinyAce_0.4.2               </span></span>
<span><span class="co">#&gt; [107] cachem_1.0.8                  GlobalOptions_0.1.2          </span></span>
<span><span class="co">#&gt; [109] stringr_1.5.0                 BiocVersion_3.17.1           </span></span>
<span><span class="co">#&gt; [111] parallel_4.3.1                miniUI_0.1.1.1               </span></span>
<span><span class="co">#&gt; [113] vipor_0.4.5                   AnnotationDbi_1.62.2         </span></span>
<span><span class="co">#&gt; [115] restfulr_0.0.15               desc_1.4.2                   </span></span>
<span><span class="co">#&gt; [117] pillar_1.9.0                  grid_4.3.1                   </span></span>
<span><span class="co">#&gt; [119] vctrs_0.6.4                   promises_1.2.1               </span></span>
<span><span class="co">#&gt; [121] BiocSingular_1.16.0           dbplyr_2.3.4                 </span></span>
<span><span class="co">#&gt; [123] iSEEu_1.12.0                  beachmat_2.16.0              </span></span>
<span><span class="co">#&gt; [125] xtable_1.8-4                  cluster_2.1.4                </span></span>
<span><span class="co">#&gt; [127] beeswarm_0.4.0                evaluate_0.23                </span></span>
<span><span class="co">#&gt; [129] GenomicFeatures_1.52.2        cli_3.6.1                    </span></span>
<span><span class="co">#&gt; [131] compiler_4.3.1                Rsamtools_2.16.0             </span></span>
<span><span class="co">#&gt; [133] rlang_1.1.1                   crayon_1.5.2                 </span></span>
<span><span class="co">#&gt; [135] ggbeeswarm_0.7.2              fs_1.6.3                     </span></span>
<span><span class="co">#&gt; [137] stringi_1.7.12                viridisLite_0.4.2            </span></span>
<span><span class="co">#&gt; [139] BiocParallel_1.34.2           munsell_0.5.0                </span></span>
<span><span class="co">#&gt; [141] Biostrings_2.68.1             lazyeval_0.2.2               </span></span>
<span><span class="co">#&gt; [143] colourpicker_1.3.0            Matrix_1.6-1.1               </span></span>
<span><span class="co">#&gt; [145] ExperimentHub_2.8.1           hms_1.1.3                    </span></span>
<span><span class="co">#&gt; [147] sparseMatrixStats_1.12.2      bit64_4.0.5                  </span></span>
<span><span class="co">#&gt; [149] KEGGREST_1.40.1               shiny_1.7.5.1                </span></span>
<span><span class="co">#&gt; [151] highr_0.10                    interactiveDisplayBase_1.38.0</span></span>
<span><span class="co">#&gt; [153] AnnotationHub_3.8.0           memoise_2.0.1                </span></span>
<span><span class="co">#&gt; [155] bslib_0.5.1                   bit_4.0.5</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="https://github.com/NajlaAbassi/" class="external-link">Najla Abassi</a>, <a href="https://federicomarini.github.io" class="external-link">Federico Marini</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
